TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/WorkspacesController.php

```diff
@@ -395,7 +395,7 @@ public function publishWorkspaceAction(Workspace $workspace)
         if (($targetWorkspace = $workspace->getBaseWorkspace()) === null) {
             $targetWorkspace = $this->workspaceRepository->findOneByName('live');
         }
-        $workspace->publish($targetWorkspace);
+        $this->publishingService->publishNodes($this->publishingService->getUnpublishedNodes($workspace), $targetWorkspace);
```

Comment:
@aertmann okay, I see. What a mess! But I agree that we should rather fix this bug first and then make sure to find a cleaner solution for the publishing mechanism. How ever it happened that we have two of them, with their individual signals :-/